python/feldera/rest/pipeline.py

```diff
@@ -35,6 +35,19 @@ def __init__(
         self.state: Optional[dict] = None

```

Comment:
There should be a distinction between the type of the local state and what is returned by some endpoints.

Local state = name and pipeline version code (to detect if there were any concurrent problems), possibly as well program_code/description/program_config/runtime_config but they would effectively be not used as version code governs them all.

No endpoint I think should use any local state, the object should always consult the direct API endpoint to get the latest value. Possibly, it can return an error if the version code it got back is not what is currently in the local state, to govern concurrent access.

Reply:
Everything contacts the backend to return state. I don't think this variable is used elsewhere. 

This state variable is a left over, I will remove it.